### PR TITLE
SLING-5866 DefaultGetServlet obtains input stream for binary even if …

### PR DESCRIPTION
**Pull request by anagarwa**
…request is a HEAD

### CODE REVIEW

File: bundles/servlets/get/src/main/java/org/apache/sling/servlets/get/impl/helpers/HeadServletResponse.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        if (stream == null) {
+            stream = new NullServletOutputStream();
+        }
+
+        return stream;
+    }
+
+    @Override
+    public PrintWriter getWriter() {
+        if (stream != null) {
```
Comment (bdelacretaz) [inline bundles/servlets/get/src/main/java/org/apache/sling/servlets/get/impl/helpers/HeadServletResponse.java:60]: the null tests are inverted between getWriter and getOutputStream,

File: bundles/servlets/get/src/main/java/org/apache/sling/servlets/get/impl/helpers/HeadServletResponse.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+    private PrintWriter writer;
+
+    public HeadServletResponse(SlingHttpServletResponse wrappedResponse) {
+        super(wrappedResponse);
+    }
+
+    @Override
+    public ServletOutputStream getOutputStream() {
+        if (writer != null) {
```
Comment (bdelacretaz) [inline bundles/servlets/get/src/main/java/org/apache/sling/servlets/get/impl/helpers/HeadServletResponse.java:47]: the null tests are inverted between getWriter and getOutputStream,

**Comment by anagarwa** [comment]
@bdelacretaz  as mentioned [here](http://docs.oracle.com/javaee/5/api/javax/servlet/ServletResponse.html#getOutputStream%28%29) "Either this method or getWriter() may be called to write the body, not both." So IMO, null test should be fine here . Moreover this is same as already done in sling [DefaultHeadServlet](https://github.com/apache/sling/blob/trunk/bundles/servlets/get/src/main/java/org/apache/sling/servlets/get/impl/DefaultHeadServlet.java#L133) . 
Please let me know if this is still a concern

**Comment by rombert** [comment]
With the upcoming migration to git this repository will be deleted. If the pull request contains data that should be re-submitted I suggest that you save the changes locally and resubmit them after the migration, which will likely happen on Monday the 9th of October.

**Comment by bdelacretaz** [comment]
It looks like this has been applied to our svn head and can be safely deleted.

**Pull request closed without merging**